### DISCUSSION TITLE
Using shell scripts different then bash (nushell specifically)

### DISCUSSION THREAD

**Discussion by ArtemChandragupta** [Aylur/ags#513] [Q&A] [answered]
Hi! We can do something like this to poll CPU usage with bash script:
```
const getUsage = Variable("", {
	poll: [
		5000,
		`bash -c "top -b -n 1 | grep 'Cpu(s)' | awk '{print $2}'"`,
		(out) => parseFloat(out).toFixed(0),
	],
});
```
But how this can be done with other command shells?

As an example, in nushell same script will be something like this:
`sys cpu | get cpu_usage | math avg`
and this is much more readable than bash solution, so I want to use this.

**Comment by Aylur** [comment, marked as answer]
Nushell also has a `-c` flag
Just swap out the bash call to nu
`nu -c "sys cpu  | get cpu_usage | math avg"`
